health-services/plan-service/src/main/java/digit/util/MdmsV2Util.java

```diff
@@ -0,0 +1,74 @@
+package digit.util;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import digit.config.Configuration;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.tracer.model.CustomException;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ObjectUtils;
+import org.springframework.web.client.RestTemplate;
+import digit.web.models.mdmsV2.*;
+
+import java.util.*;
+
+import static digit.config.ServiceConstants.*;
+
+@Slf4j
+@Component
+public class MdmsV2Util {
+
+    private RestTemplate restTemplate;
+
+    private ObjectMapper objectMapper;
+
+    private Configuration configs;
+
+    public MdmsV2Util(RestTemplate restTemplate, ObjectMapper objectMapper, Configuration configs)
+    {
+        this.restTemplate = restTemplate;
+        this.objectMapper = objectMapper;
+        this.configs = configs;
+    }
+
+    public Object fetchMdmsV2Data(RequestInfo requestInfo, String tenantId, String schemaCode)
+    {
+        StringBuilder uri = getMdmsV2Uri();
+        MdmsV2CriteriaReq mdmsV2CriteriaReq = getMdmsV2Request(requestInfo, tenantId, schemaCode);
+        Object mdmsResponseMap = new HashMap<>();
+        try{
+            mdmsResponseMap = restTemplate.postForObject(uri.toString(), mdmsV2CriteriaReq, Map.class);
```

Comment:
Why is the response being parsed as response map? MDMS v2 search returns a list of MDMS data entries.